airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/metadata.py

```diff
@@ -197,13 +198,13 @@ def _run(self, connector: Connector) -> CheckResult:
                 message=f"No upgrade deadline found for the breaking changes in {current_version}.",
             )
 
-        upgrade_deadline_datetime = datetime.strptime(upgrade_deadline, "%Y-%m-%d")
+        upgrade_deadline_datetime = upgrade_deadline.strftime(self.date_format) if isinstance(upgrade_deadline, str) else datetime.strptime(upgrade_deadline, self.date_format)
```

Comment:
This is daunting to me:
I read: call `strftime` on `upgrade_deadline` if it's a string.
`strftime` is a `datetime` object method, not a string one.
I believe this might lead to `AttributeError: 'str' object has no attribute 'strftime'`